shell/components/ChartPsp.vue

```diff
@@ -14,24 +14,71 @@ export default {
     mode: {
       type:    String,
       default: 'edit'
+    },
+
+    /**
+     * Optional title section prior checkbox
+     */
+    title: {
+      type:    String,
+      default: null
+    },
+
+    /**
+     * Cluster information
+     */
+    cluster: {
+      type:    Object,
+      default: null
     }
   },
+  data: () => {
+    return {
+      // This is required to keep the PSP visible also after unchecking the checkbox
+      hasPsp: false
+    };
+  },
   created() {
     if (!this.value.global.cattle) {
       this.$set(this.value.global, 'cattle', { psp: { enabled: false } });
     }
     if (!this.value.global.cattle.psp) {
       this.$set(this.value.global.cattle, 'psp', { enabled: false });
     }
+
+    this.hasPsp = this.value.global.cattle.psp.enabled;
   },
-  computed: { ...mapGetters({ t: 'i18n/t' }) }
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    /**
+     * Display checkbox only if contains PSP or K8S version is less than 1.25
+     */
+    hasCheckbox() {
+      const clusterVersion = this.cluster?.metadata?.annotations?.['management.cattle.io/current-cluster-controllers-version'] || '';
```

Comment:
This looks like a new way of getting the k8s version. Could the mgmt cluster `kubernetesVersion` getter be used here? eg const `clusterVersion = cluster.kubernetesVersion`